### PR TITLE
[usb,sival] Add a USB test harness for silicon exec_env

### PR DESCRIPTION
**Pull request by pamaury**
On teacup, the VBUS sense buffer is disabled by default so the device cannot detect VBUS. This PR introduces a new test harness that bootstraps the device and enable VBUS sensing.

### CODE REVIEW

**Review by alees24 — APPROVED** [review]
Thanks for this.

File: sw/host/tests/chip/usb/usb_harness.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+fn main() -> Result<()> {
+    let opts = Opts::parse();
+    opts.init.init_logging();
+    let transport = opts.init.init_target()?;
+
+    // Wait until test is running.
+    let uart = transport.uart("console")?;
+    let _ = UartConsole::wait_for(&*uart, r"Running [^\r\n]*", opts.timeout)?;
```
Comment (nbdd0121) [inline sw/host/tests/chip/usb/usb_harness.rs:35]:
```suggestion
    UartConsole::wait_for(&*uart, r"Running [^\r\n]*", opts.timeout)?;
```

File: sw/host/tests/chip/usb/BUILD [hunk 2]
```diff
@@ -36,3 +36,19 @@ rust_binary(
...
+        "usb_harness.rs",
+    ],
+    deps = [
+        ":usb",
+        "//sw/host/opentitanlib",
+        "@crate_index//:anyhow",
+        "@crate_index//:clap",
+        "@crate_index//:humantime",
+        "@crate_index//:log",
+        "@crate_index//:rusb",
```
Comment (nbdd0121) [inline sw/host/tests/chip/usb/BUILD:52]: `rusb` doesn't seem directly used?

File: sw/device/tests/BUILD [hunk 3]
```diff
@@ -3570,6 +3570,14 @@ opentitan_test(
         EARLGREY_TEST_ENVS,
         EARLGREY_SILICON_OWNER_ROM_EXT_ENVS,
     ),
+    silicon = silicon_params(
+        test_cmd = """
+            --bootstrap="{firmware}"
+            --vbus-sense-en=VBUS_SENSE_EN
+            --vbus-sense=VBUS_SENSE
```
Comment (nbdd0121) [inline sw/device/tests/BUILD:3577]: Any reason to not make them the default in usb.rs?

It's a shame that we have a ton of code duplication here, but Bazel doesn't allow function definitions in BUILD...

**Review by nbdd0121 — APPROVED** [review]

**Pull request merged**